### ISSUE TITLE
mlx-lm 0.0.4 pypi package has outdated utils.py

### ISSUE THREAD

**Issue by gladjoyhub** [ml-explore/mlx-examples#371]
In the newest mlx-lm 0.0.4 pypi package, utils.py, the yield of generate_step() has become a tuple of token and t0, but the generate() doesn't change and still treat the yield as just a token, hence cause error tuple object doesn't have .item() in tokens.append(token.item()). The mlx-examples repo on github has updated the generate() in utils.py though.

`def generate_step(
    prompt: mx.array, model: nn.Module, temp: float = 0.0, return_probability: bool = False
) -> Generator[mx.array, None, None]:
        def sample(logits: mx.array) -> Tuple[mx.array, float]:
        prop = 1
        if temp == 0:
            token = mx.argmax(logits, axis=-1)
        else:
            token = mx.random.categorical(logits * (1 / temp))
            if return_probability:
                probs = mx.softmax(logits / temp)
                prop = probs[0, token.item()]
        return token, prop

    y = prompt
    cache = None
    while True:
        logits, cache = model(y[None], cache=cache)
        logits = logits[:, -1, :]
        y, t0 = sample(logits)
        yield y, t0`

**Comment by awni** [comment]
Fixed already 😄  `pip install -U mlx-lm`
